### ISSUE TITLE
Advice on use of terms from orphaned REX ontology

### ISSUE THREAD

**Issue by schaferw** [OBOFoundry/OBOFoundry.github.io#2002]
The REX ontology has a number of terms that are pertinent to PROCO's domain but we are reluctant to just import them as-is since the REX ontology is orphaned.  There would be no mechanism for term editing etc.  Is it possible to incorporate them into PROCO giving attribution to the original work?  We would then assume responsibility for governance of the terms going forward.

**Comment by nataled** [comment]
--corrected labels and assignee--

Your question is one that is not fully settled, so far as I know. I'm therefore going to make sure this is discussed at the next OBO Foundry conference call, as it is of wide interest. The call will take place August 9th. If time is more urgent, I suggest soliciting advice from the obo-discuss@googlegroups.com mailing list.

**Comment by schaferw** [comment]
We can wait for the scheduled August 9th meeting.

**Comment by cmungall** [comment]
## Option 1

PROCO (and possibly a consortium of other groups) takes over REX and revives it. This would require the consent of the last owner, Kirill, and this should be done openly on obo-discuss to give any current users of REX the option to speak up with any concerns, but I think it should be fine. There would be an initial cost involved in setting up a modern ODK repo for REX that would make it easy for others to contribute.

## Option 2

A new ontology or new branch of PROCO is created for physico-chemical processes. REX terms are copied there as needed but with a new ID mintred, with full provenance pointing back to the source.

What you should not do is reuse the REX ID and modify/inject axioms

## General concerns

I am unclear on the scope differences between MOP and PROCO. Additionally I see PROCO is importing terms from allotrope, and including very general terms like "state" and placing them under BFO process. I think for general terms like state these should go via COB.

**Comment by pbuttigieg** [comment]
What's the status of PROCO?[ This branch](https://www.ebi.ac.uk/ols/ontologies/proco/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000015&lang=en&viewMode=All&siblings=false) looks concerning, with functions and states under processes.

**Comment by balhoff** [comment]
> What's the status of PROCO?[ This branch](https://www.ebi.ac.uk/ols/ontologies/proco/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000015&lang=en&viewMode=All&siblings=false) looks concerning, with functions and states under processes.

If you mean GO 'molecular_function', it is supposed to be under 'process'.

**Comment by schaferw** [comment]
Thanks very much for the prompt attention.  We will wait to hear the response from the REX developers.   Option 2 (limited import of relevant terms) is also our choice.

**Comment by matentzn** [comment]
I think limited import is really a problem without deprecting REX entirely. And if deprecate the whole ontology, we will lose a lot of potentially interesting terms. My preference would be to get the REX people to add you as editors, open REX up for community edits and you then importing whatever you need into PROCO straight from REX. Deprecting an ontology without replacement feels wrong as well.

**Comment by nataled** [comment]
@matentzn many ontologies have been deprecated without replacement, that's just the way of it sometimes. Terms that are needed from a deprecated ontology are typically rescued, I suppose, by those that (a) know how to maintain those terms, and (b) have a need for them. Those not rescued are left untouched for a reason--not needed, or perhaps not well formed.

As for limited import without deprecating, that is potentially a problem (hence my concern on the call about the possibility of an orphaned ontology becoming reactivated). The solution we are pursuing is exactly to deprecate, but to first ask about doing so.

**Comment by matentzn** [comment]
Yeah I reached out to the maintainers and will fill you in what they say. Actually FYI: it seems the Chebi team is responsible for REX now.

**Comment by K-r-ll** [comment]
Hello people,

What a surprise! I was approached by the ChEBI team who asked me what I wanted them to do about REX. Somehow I missed the fact that they were even responsible for maintaining it. Apparently the ChEBI team inherited the contact after the EBI deleted my email account. So... I suppose the aforementioned (in this thread) "REX developers" must be me. 

Anyway. At the time it was created in the early 2000s, there was little, if any, enthusiasm at the EBI, and not much after that either. There was some vague and non-committal interest from the Royal Society of Chemistry because at some point they used to cross-reference their publications with REX terms. The last changes to REX were made (by me) more than a decade ago which means it's about time to move on.

Sure it would be nice if anybody gave it a new life, maybe (just maybe) preserving some of its original <strike>ideas</strike> <u><strike>form</strike></u> whatever. For me, @cmungall's option 1 sounds more attractive. I also don't think it is incompatible with Option 2. Nothing should prevent the users to keep recycling the REX terms as they please.

If you have any questions about those terms themselves, feel free to ask; not sure if by now I can answer though.

**Comment by schaferw** [comment]
Thanks for the comments.  We have also heard back from CheBI on the issue.  We do not believe we have the resources to support a separate ontology from PROCO.

If REX is still indeed supported by CheBI, why is it listed as orphaned?  And can we submit term requests?  If it is not supported, we will import the terms with PROCO IRIs, noting the original source of the terms.

**Comment by schaferw** [comment]
As for the general concerns, the domain of PROCO is process chemistry, which is a branch of chemistry (including pharmaceutical chemistry) that studies the development and optimization of the production processes for chemical compounds, and the scaling up of laboratory chemical reactions into commercially viable routes. Key considerations are product quality, process robustness, economics, environmental sustainability, regulatory compliance and safety.

Although it has a distinct domain (having its own journals etc), it is fundamentally based on chemistry and leverages many terms from the relevant OBO ontologies such as CheBI, CHMO, MOP and CHEMINF.  The biggest overlap is with CHMO.  To date we have been able to simply the import CHMO terms but we are starting to ask for new CHMO terms and may seek term modifications in the future.

Having started out in ontology development through Allotrope, we perhaps lean on it too much at times.  We will investigate using COB for general terms.  (BTW Allotrope has reached out to determine if a formal relationship with OBO could be established).

**Comment by schaferw** [comment]
For option 2, we presume it is OK to assert additional or different superclasses for imported terms being careful not to violate any BFO structure or axioms.

For example glass is a polymer in CheBI which is true but we will also assert it as Material of Construction (independent continuant) subclass in the next release as compatibility of equipment with production reaction streams is a major concern in process chemistry.  (this also better aligns the term with BFO).

**Comment by K-r-ll** [comment]
Further to my above post. Mr Google brought me a thread from 2014 which sheds some light on interaction of CHMO, REX and FIX: https://code.google.com/archive/p/obo-foundry-operations-committee/issues/128

Janna Hastings said:

> We do want to keep these alive at least until we have replacement ontologies in the same domain. <...> I've just moved the sources to the ChEBI webspace as these were originally in Kirill's home drive in EBI which has since been removed as he left EBI some years back.

Curiously, or maybe not, on the top of the same thread, Colin Batchelor said:

> So I think the story is to keep REX and I'll make sure that everything in FIX has been put in CHMO with appropriate connections to OBI and then we're fine. 

So it seems that the RSC people at least intended to recycle _everything_ of FIX. However it does not look like they kept the corresponding FIX IDs, as claimed:

> There are quite a few cross-references in CHMO to FIX; Hilary was diligent about this but she may have registered them in a slightly eccentric way.

For instance, cf. [momentum-resolved Bremsstrahlung spectroscopy](https://www.ebi.ac.uk/ols/ontologies/fix/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FFIX_0000688) FIX:0000688 and [momentum-resolved bremsstrahlung spectroscopy](https://www.ebi.ac.uk/ols/ontologies/chmo/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FCHMO_0000380) CHMO:0000380

**Comment by cmungall** [comment]
On Thu, Aug 11, 2022 at 3:36 AM schaferw ***@***.***> wrote:

> For option 2, we presume it is OK to assert additional or different
> superclasses for imported terms being careful not to violate any BFO
> structure or axioms.


Many others also presume this. I think it is bad practice. But we have not
reached consensus on this. See this discussion:
https://github.com/OBOFoundry/OBOFoundry.github.io/issues/1991


> For example glass is a polymer in CheBI which is true but we will also
> assert it as Material of Construction (independent continuant) subclass in
> the next release as compatibility of equipment with production reaction
> streams is a major concern in process chemistry. (this also better aligns
> the term with BFO).


This would be incorrect, there are naturally formed glasses. Even if it
were true, I don't think this should be injected into an ontology by a 3rd
party.

If you want to use a term from CHEBI, I would start by requesting a role
assignment from CHEBI. The way CHEBI uses role is fairly fuzzy, I wouldn't
think of BFO roles, they are just useful annotations. I think chebi:glass
chebi:has_role construction role is perfectly fine and follows the same
pattern and most other CHEBI role assignments.

This all opens up a slew of really hard ontology modularization/scope
problems though.

First, we have materials in both CHEBI and ENVO
https://github.com/ebi-chebi/ChEBI/issues/3293

ENVO actually has glass
http://purl.obolibrary.org/obo/ENVO_01000480

ENVO also has process terms for things like construction
http://purl.obolibrary.org/obo/ENVO_01000996

I think there are a number of advantages to using ENVO for complex
materials, and we would be open to coming up with an ontologically sound
way to help you classify glass.

**Comment by matentzn** [comment]
@cmungall can you summarise what your opinion is on the question of whether it is acceptable to adopt terms from a non-obsolete ontology? It seems odd to me. I think REX should either be maintained as a whole, or obsoleted, but allowing other ontology groups to grabbing REX terms, axiom inject, change labels etc etc, while REX is still active, seems against OBO principles

**Comment by cmungall** [comment]
REX isn't obsolete, it's inactive. Obviously activating is the best. But
rescue via piecemeal adoption may be most pragmatic approach. The most
important thing though is a clear modular design that delineates the scope
of each ontology and how they relate

**Comment by pbuttigieg** [comment]
An observation:

Many rounds of rescue will (and has) split generic/broadly applicable content across other ontologies with narrow(er) scope. This also happens when an ontology with a narrow(er) focus decides to handle very generic content (e.g. plans), which should be maintained as its own (relatively static) ontology.

This makes the overhead importing the natural set of terms (now scattered across ontologies) a headache or intractable to import and coordinate, and also subject to idiosyncracies from the narrower foci.

"Pragmatism"

If it were easier to edit multiple ontologies and sync editor versions (syncing release cycles is non-viable at this stage), it would be much easier to co-maintain generic ontologies. That is the change that will improve (inter)operations sustainably. 

Right now, I would just create terms in ENVO (even those outside its scope) rather than dealing with handling import and integration of scattered terms that should have their own generic ontology.

**Comment by StroemPhi** [comment]
When comparing REX and MOP, one can see a great overlap between the two (e.g. catalysis or oxidation) that unfortunately is not marked with annotations in MOP. In terms of scope and orthogonality this is a problem. I also found the 2014 thread Kiril referenced in his https://github.com/OBOFoundry/OBOFoundry.github.io/issues/2002#issuecomment-1212558184 and thus assumed, since REX and FIX are orphaned on the OBO page, that MOP, RXNO and CHMO are now the place to look for terms in this scope/domain. It would definitely help to have a better comment/explanaition on what 'orphaned' means. I interpreted it as: it's listed here for backwards compability reasons but you should better not use it in new projects, as it is not maintained anymore and instead use maintained ontologies with similar scope/domain.

_Now, if REX were to be reactivated the overlap with MOP would need to be resolved, otherwise users will get really frustrated by not knowing which one to take and there would be a broken OBO rule._ 

To me, it seems that MOP would be the better place to have those terms of REX that don't have a MOP equivalent already, as @batchelorc is actively maintaing MOP, REX has many missing definitions and has less BFO structure than MOP. But importing the tems from REX, if reactivated, via ODK might also work. For the latter though, I think REX would have to be vetted regarding its subsumption hierarchy (e.g. 'oxidation' and 'reduction' being siblings of 'process', which is not bfo'process').

IMHO, I think it would be best to combine the efforts to harmonize REX and MOP to have a proper reference ontology dealing with the most basic chemical processes. A couple of month ago, Colin and I untangled [MOP and RXNO](https://github.com/rsc-ontologies/rxno/pull/44) with the goal to have both seperatly in an ODK repo and then properly import MOP into RXNO. I came quite far already but then had to switch to tasks with a higher priority unfortunately. So having MOP in a new ODK repo, might probably be easier to achieve than for REX.

If there where a scipt or tool with which we could collect the similar terms - judged by label, definitions incl def source links (often IUPAC Goldbook) - of REX on the side and MOP on the other of a table, I think the needed harmonization could be done manually with the help of domain experts/chemists. With this, we could decide to deprecate those REX terms that have a proper MOP equivalent and just import or adapt the others (if the hierarchy allows it) in/to CHMO, or at least make a SSSOM mapping between the two.

**Comment by cthoyt** [comment]
Yup I’ve made scripts like this before. Even more, I’m familiar with the overlaps between these ontologies, sort of related to Chris’s recent blog post about shadow ontologies

**Comment by cmungall** [comment]
I like the path emerging here, a unified chemical process ontology seems like the way to do

**Comment by cthoyt** [comment]
I did a lot of curation in that direction to support pybel in the past, would be very interested to be involved in or lead that effort, assuming we could do all curation in TSVs :)

**Comment by nlharris** [comment]
Is there more to be discussed here, or has the original question been answered sufficiently?
